src/view/placeholder/placeholder.jsx

```diff
@@ -1,17 +1,30 @@
 // @flow
 import React, { PureComponent } from 'react';
+import type { Placeholder as PlaceholderType } from '../../types';
 
 type Props = {|
-  height: number,
-  width: number,
+  placeholder: PlaceholderType,
 |}
 
 export default class Placeholder extends PureComponent<Props> {
   render() {
+    // We apply the margin separately to maintain margin collapsing
+    // behavior of the original element
+
+    const placeholder: PlaceholderType = this.props.placeholder;
```

Comment:
This is the core change: applying the width and height without margin, and applying the margins directly ourselves to avoid margin collapsing